File: SDKs/parliament.uk/bills/v1/python/src/sdk/models/shared/newsarticlessummarysearchresult.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from . import newsarticlessummary


@dataclass_json
@dataclass
class NewsArticlesSummarySearchResult:
    items: Optional[List[newsarticlessummary.NewsArticlesSummary]] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'items' }})
    items_per_page: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'itemsPerPage' }})
    total_results: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'totalResults' }})
    

File: SDKs/parliament.uk/bills/v1/typescript/src/sdk/models/shared/newsarticlessummarysearchresult.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { NewsArticlesSummary } from "./newsarticlessummary";



export class NewsArticlesSummarySearchResult extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=items", elemType: NewsArticlesSummary })
  items?: NewsArticlesSummary[];

  @SpeakeasyMetadata({ data: "json, name=itemsPerPage" })
  itemsPerPage?: number;

  @SpeakeasyMetadata({ data: "json, name=totalResults" })
  totalResults?: number;
}
